### PR TITLE
Add Integration test step in App creation wizard

### PR DESCRIPTION
**Pull request by karthikjeeyar**
## Fixes 

https://issues.redhat.com/browse/HACBS-815

## Description

Contains Add Integration test scenario step

## Type of change

- [x] Feature

## Screen shots / Gifs for design review 

image

## Test cases

```
Create Utils
    ✓ Should call k8sCreateResource with integration test data (3 ms)
    ✓ Should contain the display name in the annotations (1 ms)
    ✓ Should contain kubernetes formatted resource name (1 ms)
```

## How to test or reproduce?

This step can be found in Create Application wizard

## Browser conformance: 

- [x] Chrome
- [x] Firefox
- [ ] Safari
- [ ] Edge

### CODE REVIEW

**Comment by karthikjeeyar** [comment]
cc: @Ranelim @MariaLeonova

File: src/hacbs/components/ImportForm/submit-utils.ts [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+import { sanitizeName } from '../../../utils/create-utils';
+import {
+  IntegrationTestScenarioGroupVersionKind,
+  IntegrationTestScenarioModel,
+} from '../../models/hacbs';
+import { IntegrationTestScenarioKind } from '../../types/coreBuildService';
+import { FormValues, IntegrationTestFormValues } from './types';
+
+export enum IntegrationTestAnnotations {
+  DISPLAY_NAME = 'app.kubernetes.io/display-name',
```
[the remark this answers is not in the document]
Comment (christianvogt) [reply src/hacbs/components/ImportForm/submit-utils.ts:12]: nm. These are recommended labels and we need an annotation for the display name.
Ignore my comment and use what you have.

File: src/hacbs/components/ImportForm/IntegrationTestSection.tsx [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        <Text component={TextVariants.p}>
+          <a href="#">Learn more about setting up an integration test pipeline</a>
+        </Text>
+      </TextContent>
+      <br />
+      <Form isWidthLimited>
+        <InputField
+          label="Display name"
+          name="integrationTest.name"
+          placeholder="Enter your source"
```
[the remark this answers is not in the document]
Comment (karthikjeeyar) [reply src/hacbs/components/ImportForm/IntegrationTestSection.tsx]: oops, Its display name placeholder :) , updated it.

**Comment by MariaLeonova** [comment]
@karthikjeeyar Looks great! One small thing I noticed: there's a typo here:
Screenshot 2022-08-05 at 13 38 10

**Comment by Ranelim** [comment]
Thank you @karthikjeeyar !
The step looks excellent. 
We need to add an external link icon to the "Learn more"  link above.

@karthikjeeyar @christianvogt 
Will we make the 'Cancel' button a 'Go to application' once we pass a step where we created something? I think we landed on creating the app after the first step.
Will that be tackled in a separate PR?

Thank you!

**Comment by christianvogt** [comment]
@Ranelim  see https://github.com/openshift/hac-dev/pull/170

**Comment by karthikjeeyar** [comment]
E2E tests are passing locally

image

**Comment by karthikjeeyar** [comment]
Thanks @Ranelim @MariaLeonova Updated the PR.

**Comment by christianvogt** [comment]
@karthikjeeyar @rohitkrai03 
If i'm on the integration page and fill it out, then go back to the previous page and then forward to the next page, the form seems to get submitted

File: src/hacbs/components/ImportForm/useImportSteps.tsx [hunk 3]
```diff
@@ -58,6 +61,25 @@ export const buildStep = (): FormikWizardStep => ({
...
+  onSubmit: createAppIntegrationTest,
+});
+
+export const environmentStep = (): FormikWizardStep => ({
+  id: 'environment',
+  name: 'Manage Environment',
+  component: <></>,
+  canJumpTo: false,
   nextButtonText: 'Done! Go to app',
 });
```
Comment (christianvogt) [inline src/hacbs/components/ImportForm/useImportSteps.tsx:84]: Is this empty step still necessary to include right now? 
If so, add `cancelButtonText: 'Go to application',`

Comment (karthikjeeyar) [reply src/hacbs/components/ImportForm/useImportSteps.tsx]: Updated

**Comment by karthikjeeyar** [comment]
> @karthikjeeyar @rohitkrai03
If i'm on the integration page and fill it out, then go back to the previous page and then forward to the next page, the form seems to get submitted

First time landing on the step and also clicking on the current step link is submitting the form , I have logged an issue for this https://github.com/patternfly-labs/formik-pf/issues/19

**Comment by christianvogt** [comment]
image

**Pull request merged**